packages/components/src/components/FileUploader/FileUploader.tsx

```diff
@@ -63,13 +66,19 @@ const FileUploader = React.forwardRef<HTMLDivElement, FileUploaderProps>(
       onCancel,
       onRemove,
       disabled = false,
+      required,
     },
     ref
   ) => {
     const isMobile = useDown("md");
     const status = getStatus({ progress, fileUrl, errorMessage });
     const isMobileUploading = isMobile && status === "loading";
     const isUploadedWithoutName = status === "success" && !fileName;
+    const hasFile = fileName || fileUrl;
+    const shouldShowChildren =
+      status === "waiting" || (required && errorMessage);
```

Comment:
Do we need to be checking `required` and `errorMessage` here? Just checking if it's an error state and that hasFile is false would be more comprehensive.